openmrs-19/src/main/resources/webapp/js/controllers.js

```diff
@@ -0,0 +1,156 @@
+
+        $scope.configs = undefined;
+        $scope.selectedConfig = undefined;
+        $scope.configOutdated = false;
+        $scope.newPatientIdentifierType={};
+
+        $scope.getDefault = function () {
+            var i;
+            for (i = 0; i < $scope.configs.configs.length; i += 1) {
+                if ($scope.configs.configs[i].name === $scope.configs.defaultConfigName) {
+                    return $scope.configs.configs[i];
+                }
+            }
+            return undefined;
+        };
+
+         $scope.getConfigs = function () {
+             OpenMRSConfig.get(
+             function success(data) {
+                 $scope.configs = data;
+                 $scope.configs.configs.patientIdentifierTypeNames=[];
+                 $scope.configOutdated = false;
+                 if ($scope.selectedConfig === undefined) {
+                    $scope.selectedConfig = $scope.getDefault();
+                 }
+            });
+        };
+
+        $scope.getConfigs();
+
+        $scope.addConfig = function () {
+            $scope.selectedConfig = {
+                'name':'',
+                'openMrsUrl':'',
+                'username':'',
+                'password':'',
+                'motechPatientIdentifierTypeName':'',
+                'patientIdentifierTypeNames':[]
+            };
+        };
+
+        $scope.hasValue = function(prop) {
+            return $scope.selectedConfig[prop] !== null &&
+                $scope.selectedConfig[prop] !== undefined &&
+                $scope.selectedConfig[prop] !== "";
+        };
+
+        $scope.draftChanged = function() {
+            $scope.configOutdated = true;
+        };
+
+        $scope.selectChanged = function() {
+            $scope.configOutdated = false;
+        };
+
+        $scope.addPatientIdentifierType = function() {
+            $scope.selectedConfig.patientIdentifierTypeNames.push($scope.newPatientIdentifierType.name);
+            $scope.newPatientIdentifierType.name = "";
+            $scope.configOutdated = true;
+        };
+
+        $scope.removePatientIdentifierType = function(patientIdentifierTypeName) {
+            $scope.selectedConfig.patientIdentifierTypeNames.splice($scope.selectedConfig.patientIdentifierTypeNames.indexOf(patientIdentifierTypeName), 1);
+            $scope.configOutdated = true;
+        };
+
+        $scope.validateConfig = function() {
+            return $scope.hasValue('name')
+                && $scope.hasValue('openMrsUrl')
+                && $scope.hasValue('username')
+                && $scope.hasValue('password')
+                && $scope.hasValue('motechPatientIdentifierTypeName');
+        };
+
+        $scope.saveAllowed = function() {
+            return $scope.validateConfig()
+                && $scope.configOutdated;
+        };
+
+        $scope.isNewConfig = function() {
+            for (var i=0; i<$scope.configs.configs.length; i++) {
```

Comment:
is declaring var in this line valid? you declared it outside the for declaration previously